src/runtime_src/core/pcie/windows/shim.h

```diff
@@ -19,6 +19,105 @@
 
 #include "xrt.h"
 #include "core/common/xrt_profiling.h"
+#include "boost/any.hpp"
+#include <boost/property_tree/ptree.hpp>
+#include "core/pcie/driver/windows/include/XoclUser_INTF.h"
+
+typedef enum {
```

Comment:
I am not understand the design.  Why are all these enums added here.
The idea with device_core/device_windows is that all query commands translate into one or more ioctl calls.  I don't understand the apparent code duplication into shim itself.  shim.cpp should define the functions that make the new ioctl calls only.  The proper shim function to call should be determined at a different level.

Please document your design if I am missing something.

Reply:
These enums were added temperarily to get output in xbutil query through already existing IOCTLs. Since, we need to define a IOCTL for each Query Request, so it will need support from driver side also. Instead, we added these enums temporarily to show the flow is working, but these will be removed with proper design next.

We will remove these enums and will define mapping of IOCTLs to each and every QueryRequest (QR).